modules/dcache/src/main/java/org/dcache/pool/statistics/IoStatistics.java

```diff
@@ -0,0 +1,205 @@
+
+        _totalReadTime += readTime;
+        _readRequestNum ++;
+        _readBytes += readBytes;
+    }
+
+    /**
+     *
+     * Update all statistics depending on write requests.
+     *
+     * @param  writeBytes
+     *         The number of written bytes; must be non-negative
+     *
+     * @param  writeTime
+     *         The duration of the write request in milliseconds; must be non-negative
+     */
+
+    public void updateWrite(long writeBytes, long writeTime){
+        float writeSpeed = calculateSpeed(writeBytes, writeTime);
+        _writeSpeeds.add(writeSpeed);
+
+        _minWriteSpeed = (_writeRequestNum == 0) ? writeSpeed : Math.min(writeSpeed, _minWriteSpeed);
+        _maxWriteSpeed = (_writeRequestNum == 0) ? writeSpeed : Math.max(writeSpeed, _maxWriteSpeed);
+        _avgWriteSpeed = (_writeRequestNum == 0) ? writeSpeed : calculateNewAvg(_writeRequestNum,_avgWriteSpeed, writeSpeed);
+        _95WriteSpeed = (_writeRequestNum == 0) ? writeSpeed : calculate95Percentile((_writeRequestNum + 1), _writeSpeeds);
+
+        _totalWriteTime += writeTime;
+        _writeRequestNum ++;
+        _writtenBytes += writeBytes;
+    }
+
+    /**
+     *
+     * Calculate the speed of a request.
+     *
+     * @param  bytes
+     *         The number of processed bytes; must be non-negative
+     *
+     * @param  time
+     *         The duration of the request in milliseconds; must be non-negative
+     *
+     * @return The speed of the process in bytes per millisecond.
+     */
+
+    private float calculateSpeed(long bytes, long time){
+        return (float) bytes / (float) time;
+    }
+
+    /**
+     *
+     * Calculate the new average of the speed of all read or write requests.
+     *
+     * @param  requestNum
+     *         The number of read or write requests so far; must be non-negative
+     *
+     * @param  oldAvg
+     *         The old average of the speed of all read or write requests; must be non-negative
+     *
+     * @param  newSpeed
+     *         A new speed value of a read or write request; must be non-negative
+     *
+     * @return The new average of the speed of all read or write requests.
+     */
+
+    private float calculateNewAvg(long requestNum, float oldAvg, float newSpeed){
+        return (requestNum * oldAvg + newSpeed) / (requestNum + 1);
+    }
+
+    /**
+     *
+     * Calculate the 95% percentile of the speed of all read or write requests.
+     *
+     * @param  requestNum
+     *         The number of read or write requests so far; must be non-negative
+     *
+     * @param  speeds
+     *         All speeds of all read or write requests so far.
+     *
+     * @return The calculated 95% percentile of the speed of all read or write requests.
+     */
+
+    private float calculate95Percentile(long requestNum, ArrayList<Float> speeds){
+        Collections.sort(speeds);
```

Comment:
This is not essential, but you do realize it is not efficient to sort the list each time you recalculate.

It would be better if you could maintain the sort order on insert/add.  But  this is just a suggestion.